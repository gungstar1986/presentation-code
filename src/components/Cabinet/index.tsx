import React from 'react';
import { observer } from 'mobx-react-lite';

import { useUserStore } from 'hooks/storesHooks/useUserStore';
import { useBeerStore } from 'hooks/storesHooks/useBeerStore';

export const CabinetMainComponent: React.FC = observer(() => {
  const { user, setUserLogout } = useUserStore();
  const { setInitialValues } = useBeerStore();

  if (!user) return null;

  const handleLogout = () => {
    setUserLogout();
    setInitialValues();
  };

  const { name, email, role } = user;

  return (
    <div className="main-content-container">
      <div className="form-container">
        <div className="login">
          <div className="login-body-item">
            <div className="logout-item-title">User Information</div>
            <div className="logout-item-container">
              <div className="logout-item">Name:</div>
              <div className="logout-item description">{name}</div>
            </div>
            <div className="logout-item-container">
              <div className="logout-item">Email:</div>
              <div className="logout-item description">{email}</div>
            </div>
            <div className="logout-item-container">
              <div className="logout-item">Role:</div>
              <div className="logout-item description">{role}</div>
            </div>
            <div className="button-container">
              <button className="btn-primary mod" type="button" onClick={handleLogout}>
                Logout
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
});
