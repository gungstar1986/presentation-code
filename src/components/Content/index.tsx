import React, { useEffect } from 'react';
import { Route } from 'react-router-dom';
import { observer } from 'mobx-react-lite';

import { useBeerStore } from 'hooks/storesHooks/useBeerStore';

import { BeersList } from 'components/Content/BeersList';
import { CONTENT, BEER_FROM_COLLECTION } from 'constants/routes';
import { BeerInfoCommon } from 'components/Content/BeerInfoCommon';

export const ContentMainComponent: React.FC = observer(() => {
  const { fetchBeerList, beerList } = useBeerStore();

  useEffect(() => {
    if (!beerList) fetchBeerList();
  }, []);

  if (!beerList) return null;

  return (
    <div className="main-content-container">
      <Route path={CONTENT} component={BeersList} />
      <Route exact path={BEER_FROM_COLLECTION} component={BeerInfoCommon} />
    </div>
  );
});
