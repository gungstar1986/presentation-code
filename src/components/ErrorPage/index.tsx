import React from 'react';

import { useAuthCheck } from 'hooks/customHooks/useAuthCheck';

export const ErrorPage = () => {
  useAuthCheck();

  return (
    <div className="main-content-container">
      <div className="form-container">
        <div className="login">
          <div className="not-support-wrapper">
            <div className="wrapper">
              <div className="warning-icon"></div>
              <h1 className="title">Wrong way!</h1>
              <p className="text">We're sorry, this page does not exist.</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
