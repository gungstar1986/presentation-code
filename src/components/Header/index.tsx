import React from 'react';

export const Header: React.FC = () => {
  return <header className="header">Header</header>;
};
