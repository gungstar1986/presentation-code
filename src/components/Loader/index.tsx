import React, { useContext } from 'react';
import { observer } from 'mobx-react-lite';

import { RootStoreContext } from 'store';

import animation from 'assets/loader.svg';

import 'components/Loader/style.scss';

export const Loader: React.FC = observer(({ children }) => {
  const stores: { [key: string]: any } = useContext(RootStoreContext);

  const isPending = Object.keys(stores).find((store) => stores[store].isPending);

  return (
    <div className="app">
      {isPending && (
        <div className="spinner">
          <img src={animation} alt="loading" />
        </div>
      )}
      {children}
    </div>
  );
});
