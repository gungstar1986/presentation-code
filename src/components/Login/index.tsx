import React from 'react';
import { observer } from 'mobx-react-lite';

import { useUserStore } from 'hooks/storesHooks/useUserStore';
import { useRedirectToCabinet } from 'hooks/customHooks/useRedirectToCabinet';

import { Login } from 'layouts/Login';
import { User } from 'store/user/interfaces';
import { FormikContextProvider } from 'components/common/FormikComponent';
import { loginFormicSchema, loginFormikData } from 'components/Login/constants';

export const LoginMainComponent: React.FC = observer(() => {
  const { checkUserIsAuth } = useUserStore();

  useRedirectToCabinet();

  const handleSubmit = (data: User) => {
    checkUserIsAuth(data);
  };

  return (
    <div className="main-content-container">
      <div className="form-container">
        <FormikContextProvider accept={handleSubmit} formikData={loginFormikData} formikSchema={loginFormicSchema}>
          <Login />
        </FormikContextProvider>
      </div>
    </div>
  );
});
