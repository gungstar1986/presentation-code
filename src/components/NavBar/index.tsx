import React, { useEffect, useState } from 'react';
import { NavLink, useLocation } from 'react-router-dom';

import { useUserStore } from 'hooks/storesHooks/useUserStore';

import { CABINET, CONTENT, LOGIN } from 'constants/routes';

export const Navbar: React.FC = () => {
  const [location, setLocation] = useState<string>(null);

  const { user } = useUserStore();
  const { pathname } = useLocation();

  const handleLocation = (event: React.MouseEvent, destination: string) => {
    setLocation(pathname);
    if (location.includes(destination)) event.preventDefault();
  };

  useEffect(() => {
    setLocation(pathname);
  }, [pathname]);

  return (
    <div className="main-navbar-container">
      {!user && (
        <div className="navbar-item-container">
          <NavLink onClick={(e) => handleLocation(e, LOGIN)} activeClassName="active-link" to={LOGIN}>
            <span className="navbar-link-text">LOGIN</span>
          </NavLink>
        </div>
      )}

      {user && (
        <>
          <div className="navbar-item-container">
            <NavLink onClick={(e) => handleLocation(e, CABINET)} activeClassName="active-link" to={CABINET}>
              <span className="navbar-link-text">CABINET</span>
            </NavLink>
          </div>
          <div className="navbar-item-container">
            <NavLink onClick={(e) => handleLocation(e, CONTENT)} activeClassName="active-link" to={CONTENT}>
              <span className="navbar-link-text">CONTENT</span>
            </NavLink>
          </div>
        </>
      )}
    </div>
  );
};
