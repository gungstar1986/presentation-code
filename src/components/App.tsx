import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import { ToastContainer } from 'react-toastify';

import { RootStoreProvider } from 'store';
import { Header } from 'components/Header';
import { Loader } from 'components/Loader';
import { Navbar } from 'components/NavBar';
import { PrivatRoute } from 'routes/PrivatRoute';
import { LoginMainComponent } from 'components/Login';
import { ErrorPage } from 'components/ErrorPage';
import { CabinetMainComponent } from 'components/Cabinet';
import { ContentMainComponent } from 'components/Content';
import { CABINET, CONTENT, LOGIN, ROOT } from 'constants/routes';

import 'react-toastify/dist/ReactToastify.css';

export const App: React.FC = () => {
  return (
    <RootStoreProvider>
      <ToastContainer />
      <Loader>
        <Header />
        <Navbar />
        <Switch>
          <PrivatRoute path={LOGIN} component={LoginMainComponent} />
          <PrivatRoute path={CONTENT} component={ContentMainComponent} />
          <PrivatRoute path={CABINET} component={CabinetMainComponent} />
          <Redirect exact from={ROOT} to={LOGIN} />
          <Route component={ErrorPage} />
        </Switch>
      </Loader>
    </RootStoreProvider>
  );
};
