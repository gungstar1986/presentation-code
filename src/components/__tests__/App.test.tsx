import React from 'react';

import { render } from '@testing-library/react';

import { App } from 'components/App';
import { BrowserRouter } from 'react-router-dom';

describe('App component', () => {
  it('should render all field', () => {
    const component = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
