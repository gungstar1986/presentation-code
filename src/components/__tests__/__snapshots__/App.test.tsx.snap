// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render all field 1`] = `
<body>
  <div>
    <div
      class="Toastify"
    />
    <div
      class="app"
    >
      <header
        class="header"
      >
        Header
      </header>
      <div
        class="main-navbar-container"
      >
        <div
          class="navbar-item-container"
        >
          <a
            aria-current="page"
            class="active-link"
            href="/login"
          >
            <span
              class="navbar-link-text"
            >
              LOGIN
            </span>
          </a>
        </div>
      </div>
      <div
        class="main-content-container"
      >
        <div
          class="form-container"
        >
          <form>
            <div
              class="login"
            >
              <div
                class="login-body-item"
              >
                <div
                  class="form-control"
                >
                  <label
                    class="input-label"
                    for="email"
                  >
                    Email
                  </label>
                  <input
                    class="input"
                    id="email"
                    name="email"
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-control"
                >
                  <label
                    class="input-label"
                    for="password"
                  >
                    password
                  </label>
                  <input
                    class="input"
                    id="password"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  class="checkbox"
                >
                  <input
                    class="checkbox-input"
                    id="rememberMe"
                    name="rememberMe"
                    type="checkbox"
                  />
                  <label
                    class="checkbox-label"
                    for="rememberMe"
                  >
                    <span
                      class="checkbox-text"
                    >
                      Remember me?
                    </span>
                  </label>
                </div>
                <button
                  class="btn-primary mod"
                  type="submit"
                >
                  Вход
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</body>
`;
